src/Core/Settings.cpp

```diff
@@ -5951,6 +5951,9 @@ This only affects operations performed on the client side, in particular parsing
 Normally this setting should be set in user profile (users.xml or queries like `ALTER USER`), not through the client (client command line arguments, `SET` query, or `SETTINGS` section of `SELECT` query). Through the client it can be changed to false, but can't be changed to true (because the server won't send the settings if user profile has `apply_settings_from_server = false`).
 
 Note that initially (24.12) there was a server setting (`send_settings_to_client`), but latter it got replaced with this client setting, for better usability.
+)", 0) \
+    DECLARE(Bool, object_storage_treat_key_related_wildcards_as_star, false, R"(
```

Comment:
Three options here:

1. make it the default behavior, not behind a setting
2. make the setting on by default
3. keep it off by default